### PR TITLE
Feature/tile and unit factory

### PR DESCRIPTION
**Pull request by karthshen**

### CODE REVIEW

File: gatgsxw2.kbg.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿pick 7156c4a Extended Unit, Comparable for EntityId, Tile class
```
Comment (gdscheele) [inline gatgsxw2.kbg.txt]: can you remove this file?

File: src/main/game/entities/factories/UnitFactory.java [hunk 2]
```diff
@@ -23,16 +41,44 @@ public UnitFactory(){
...
-                return new Colonist(unitStatistics.get(unit), location);
-            case EXPLORER:
-                return new Explorer(unitStatistics.get(unit), location);
+            case COLONIST:{
+                try{
+                    newColonistId = this.colonistIdManager.getNewId();
+                    entityId = new EntityId(playerId, EntityTypeEnum.UNIT, EntitySubtypeEnum.COLONIST, newColonistId );
+                    return new Colonist(unitStatistics.get(unit), location, entityId);
+                } catch (IdLimitExceededException e) {
+                    throw new ColonistLimitExceededException("Colonist limit is reached, can't add new colonist");
```
Comment (gdscheele) [inline src/main/game/entities/factories/UnitFactory.java]: nice 👍

File: src/main/game/gameboard/Tile.java [hunk 3]
```diff
@@ -2,203 +2,107 @@
...
-//        return false;
-//    }
+    }
+
+    public boolean containsStructure(){
+        return (structure!=null);
+    }
 
+    public boolean getContainUnit(){
+        if(units.isEmpty()== true)
```
Comment (gdscheele) [inline src/main/game/gameboard/Tile.java]: could just be return !units.isEmpty();

Comment (randygbrooks) [inline src/main/game/gameboard/Tile.java]: return (!units.isEmpty())

**Comment by gdscheele** [comment]
did a manual gradle test since travis is down. Can confirm it passes.

**Pull request merged**